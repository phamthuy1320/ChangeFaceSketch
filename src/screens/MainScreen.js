import React from 'react';
import {
    View,
    Text,
    StyleSheet
}from 'react-native';
import {ButtonRect} from './commons';
import AntDesign from 'react-native-vector-icons/AntDesign';
import Fontisto from 'react-native-vector-icons/Fontisto';
// import {useNavigation} from '@react-navigation/native';
import {BGCOLOR, BUTTON} from './commons/Colors';

const BrandBox = () =>{
    const str = 'How do you want to\n\nget starting?'
    return(
        <View style = {styles.brandbox}>
            <Text style = {st_brandbox.text}>{str}</Text>
        </View>
    )
}

const ButtonBox = ({navigation}) =>{
    // const navigation = useNavigation()
    return (
        <View style = {styles.buttonBox}>
            <ButtonRect 
                onPress = {()=>navigation.navigate('Camera')}
                color = {BUTTON.inMain.camera.bg}
                title = 'Camera'
                colorText = {BUTTON.inMain.camera.text}
                icon = {<AntDesign name = 'camera' size = {50} color={BUTTON.inMain.camera.text}/>}
                row
            />
            <ButtonRect
                onPress = {()=>navigation.navigate('Library')}
                title = 'Library'
                colorText = {BUTTON.inMain.library.text}
                color = {BUTTON.inMain.library.bg}
                icon = {<Fontisto name = 'photograph' size = {50} color={BUTTON.inMain.library.text}/>}
                row
            />
        </View>
    )
}
const IntroductionBox = () =>{
    const str = 'Pham Thi Thuy K62 UET'
    return(
        <View style = {styles.introducts}>
            <Text style = {st_introductionbox.text}>{str}</Text>
        </View>
    )
}
const MainScreen = ({navigation}) =>{
    // const navigation = useNavigation()
    return(
        <View style = {styles.container}>
        {/* <ImageBackground source ={{uri:im_background}} style = {styles.container} resizeMode = 'stretch'> */}
            <BrandBox/>
            {/* <View style = {styles.brandbox} /> */}
            <ButtonBox navigation={navigation}/>
            <IntroductionBox/>
        {/* </ImageBackground> */}
        </View>
    )
}

export default MainScreen;

const styles = StyleSheet.create({
    container:{
        backgroundColor:BGCOLOR,
        flex:1,
        justifyContent:'space-between',
        padding:10,
    },
    brandbox: {
        flex:1/3,
        justifyContent:'center',
        alignContent:'center',
        alignItems:'flex-end',
        marginHorizontal:25,
    },
    buttonBox:{
        flex:1/3,
        padding:25,
        justifyContent:'center',
    },
    introduction:{
        flex:1/3
    }
})

const st_brandbox = StyleSheet.create({
    text:{
        fontSize:35,
        color:'#fff',
        textAlign:'right'
    }
})
const st_introductionbox = StyleSheet.create({
    text:{
        textAlign:'center',
        fontStyle:'italic',
        color:BUTTON.inMain.camera.text
    }
})