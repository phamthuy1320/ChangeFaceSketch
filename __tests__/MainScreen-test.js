import React from 'react';
import renderer from 'react-test-renderer';
import MainScreen from '../src/screens/MainScreen';

test('Main Screen', () => {
  const tree = renderer.create(<MainScreen/>).toJSON();
  expect(tree).toMatchSnapshot();
});