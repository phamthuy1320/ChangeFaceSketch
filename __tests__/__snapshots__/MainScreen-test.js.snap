// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Screen 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#f1c3aa",
      "flex": 1,
      "justifyContent": "space-between",
      "padding": 10,
    }
  }
>
  <View
    style={
      Object {
        "alignContent": "center",
        "alignItems": "flex-end",
        "flex": 0.3333333333333333,
        "justifyContent": "center",
        "marginHorizontal": 25,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 35,
          "textAlign": "right",
        }
      }
    >
      How do you want to

get starting?
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 0.3333333333333333,
        "justifyContent": "center",
        "padding": 25,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderRadius": 25,
          "flexDirection": "row",
          "height": 90,
          "justifyContent": "space-around",
          "marginVertical": 10,
          "opacity": 1,
          "paddingHorizontal": 10,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#9f4347",
              "fontSize": 50,
            },
            undefined,
            Object {
              "fontFamily": "anticon",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <Text
        style={
          Array [
            Object {
              "textAlign": "center",
            },
            Object {
              "color": "#9f4347",
              "fontSize": 22,
            },
          ]
        }
      >
        Camera
      </Text>
      <View />
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderRadius": 25,
          "flexDirection": "row",
          "height": 90,
          "justifyContent": "space-around",
          "marginVertical": 10,
          "opacity": 1,
          "paddingHorizontal": 10,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#9f4347",
              "fontSize": 50,
            },
            undefined,
            Object {
              "fontFamily": "Fontisto",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <Text
        style={
          Array [
            Object {
              "textAlign": "center",
            },
            Object {
              "color": "#9f4347",
              "fontSize": 22,
            },
          ]
        }
      >
        Library
      </Text>
      <View />
    </View>
  </View>
  <View>
    <Text
      style={
        Object {
          "color": "#9f4347",
          "fontStyle": "italic",
          "textAlign": "center",
        }
      }
    >
      Pham Thi Thuy K62 UET
    </Text>
  </View>
</View>
`;
